File: app/src/main/java/me/imzack/app/ender/injector/component/PlanDetailComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.PlanDetailPresenterModule;
import me.imzack.app.ender.injector.scope.ActivityScope;
import me.imzack.app.ender.view.activity.PlanDetailActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlanDetailPresenterModule.class)
public interface PlanDetailComponent {

    void inject(PlanDetailActivity planDetailActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/PlanDetailComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.PlanDetailPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.PlanDetailActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlanDetailPresenterModule::class))
interface PlanDetailComponent {

    fun inject(planDetailActivity: PlanDetailActivity)
}
